src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -33,7 +33,7 @@ class Group(
     * @return None if the app was not found or non empty option with app.
     */
   def app(appId: PathId): Option[AppDefinition] = {
-    apps.get(appId) orElse group(appId.parent).flatMap(_.apps.get(appId))
+    apps.get(appId) orElse group(appId.parent.asAbsolutePath).flatMap(_.apps.get(appId))
```

Comment:
I'm not really sure about this. We should either checm that appId starts with a '/', or check that is has the "absolute" flag, or something. But simply casting it with "asAbsolutePath" sits kind of wrong with me. There's a reason that method is already deprecated. 

Maybe we can add an override 
``` def app(appId: AbsolutePathId): Option[AppDefinition]```
and add a logger warning and deprecate this one? That would make it more obvious that we need to fix the rest of this stuff.

Reply:
App ids are always absolute. I started changing the type but it was way to much to land before my vacation. For now we have to live with it until `asAbsolutePath` is removed.